java-checks/src/main/java/org/sonar/java/checks/StaticMethodCheck.java

```diff
@@ -99,6 +98,14 @@ public void visitIdentifier(IdentifierTree tree) {
     if ("class".equals(tree.name()) || methodReferences.isEmpty()) {
       return;
     }
+    if (parentIs(tree, Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree parent = (MemberSelectExpressionTree) tree.parent();
+      // Exclude identifiers used in member select, except for instance creation
+      // New class may use member select to denote an inner class
+      if (tree == parent.identifier() && !parentIs(parent, Tree.Kind.NEW_CLASS)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)